### ISSUE TITLE
Write errors to stderr

### ISSUE THREAD

**Issue by thomas-zahner** [lycheeverse/lychee#984]
Currently some errors are written to stdout instead of stderr.

Steps to reproduce:
1. Create a file `a.md` with the following content: `My URL: https://a-very-invalid-and-non-existing-domain.org`
2. Run `lychee a.md --format json --no-progress | cat`

The output now contains not only the expected JSON, but also the error message `✗ [ERR] https://a-very-invalid-and-non-existing-domain.org/ | Failed: Network error: dns error: no record found for Query { name: Name("a-very-invalid-and-non-existing-domain.org.ost.ch."), query_type: AAAA, query_class: IN }`

Expected behaviour:

It should behave the same as with `lychee a.md --format json --no-progress -o out`. But instead of writing to the specified file it should write to stdout. This way we could pipe the output to other programs as expected. For example piping into `jq` ([jq](https://stedolan.github.io/jq/manual/)) doesn't work because of this bug: `lychee a.md --format json --no-progress | jq`

**Comment by mre** [comment]
Thanks for reporting. Attempted to fix that in #990. I might have missed a few cases, so please let me know if it works for you.

**Comment by mre** [comment]
## Test Command

```bash
echo 'https://a-very-invalid-and-non-existing-domain.org' | cargo run -- --format json --no-progress - | jq
```

## Before

```
parse error: Invalid numeric literal at line 1, column 4
```

## This PR

```
✗ [ERR] https://a-very-invalid-and-non-existing-domain.org/ | Failed: Network error: dns error: no record found for Query { name: Name("a-very-invalid-and-non-existing-domain.org.fritz.box."), query_type: AAAA, query_class: IN }
{
  "detailed_stats": false,
  "total": 1,
  "successful": 0,
  "unknown": 0,
  "unsupported": 0,
  "timeouts": 0,
  "redirects": 0,
  "excludes": 0,
  "errors": 1,
  "cached": 0,
  "success_map": {},
  "fail_map": {
    "stdin": [
      {
        "url": "https://a-very-invalid-and-non-existing-domain.org/",
        "status": "Failed: Network error"
      }
    ]
  },
  "excluded_map": {}
}
```

Note that `jq` properly handles the input, which means that `✗ [ERR] ...` is not printed to `stdout` anymore.
